### PR TITLE
Support mining empty shards

### PR DESCRIPTION
**Pull request by syntrust**
The changes aim to support shards being mined with no related `info` stored in the contract. This happens when data files have done "fill empty" but no corresponding blobs in the network. 
1. in `checkInclusive()` handle empty data and hash
2. in `_rewardMiner()` handle the case when `lastKvIdx` ==0

Passed miner unit tests with empty filled and real data blobs.

### CODE REVIEW

File: contracts/StorageContract.sol [hunk 1]
```diff
@@ -164,14 +164,14 @@ abstract contract StorageContract is DecentralizedKV {
         // Mining is successful.
         // Send reward to coinbase and miner.
         MiningLib.MiningInfo storage info = infos[shardId];
-        uint256 lastShardIdx = (lastKvIdx - 1) >> shardEntryBits;
+        uint256 lastShardIdx = lastKvIdx > 0 ? (lastKvIdx - 1) >> shardEntryBits : 0;
         uint256 reward = 0;
         if (shardId < lastShardIdx) {
             reward = _paymentIn(storageCost << shardEntryBits, info.lastMineTime, minedTs);
         } else if (shardId == lastShardIdx) {
             reward = _paymentIn(storageCost * (lastKvIdx % (1 << shardEntryBits)), info.lastMineTime, minedTs);
```
Comment (qzhodl) [inline contracts/StorageContract.sol:172]: lastMineTime in shard 0 should also be initialized by startTime in the constructor?

File: contracts/EthStorageContract.sol [hunk 2]
```diff
@@ -122,6 +122,9 @@ contract EthStorageContract is StorageContract, Decoder {
         uint256 decodedData,
         bytes memory peInput
     ) public view returns (bool) {
+        if (decodedData == 0 && dataHash == 0x0) {
```
Comment (qzhodl) [inline contracts/EthStorageContract.sol:125]:
```suggestion
        if (dataHash == 0x0) {
            if (decodedData == 0) {
                return true;
            } else {
                return false;
            }
        }
```

Comment (syntrust) [reply contracts/EthStorageContract.sol:125]: It's basically like this?
```suggestion
        if (dataHash == 0x0) {
            return decodedData == 0;
        }
```

**Comment by syntrust** [comment]
Gas evaluation of mining transactions based on this change:
## Total gas used by mine() with 2 random checks

|is shard empty|first tx|later tx|
| ---- | ---- | ---- |
|yes|629,602|577,868|
|no|714,608|685,699|

## Break down
|method|gas used|
| ---- | ---- |
|checkInclusive|81,566|
|decodeSample|250,269

**Review by qzhodl — APPROVED** [review]

**Review by qizhou — APPROVED** [review]

**Pull request merged**